core/transaction-pool/graph/src/base_pool.rs

```diff
@@ -226,6 +226,18 @@ impl<Hash: hash::Hash + Member + Serialize, Ex: ::std::fmt::Debug> BasePool<Hash
 		self.future.all()
 	}
 
+	/// Return ready transactions given list of hashes.
+	pub fn by_hash(&self, hashes: impl IntoIterator<Item=Hash> + Clone) -> Vec<Option<Arc<Transaction<Hash, Ex>>>> {
```

Comment:
Can we not use `hashes: Iter<Hash>`? Looking into the other functions, the hash is always used as reference. This could prevent a lot of extra allocations.

Reply:
Do you mean `Iterator<Item = &'a Hash>`?
1. It adds quite some noise with lifetimes (and they are unforutnatelly required)
2. IT's not so easy to convert `Iterator<Item=Hash>` into `Iterator<Item=&Hash>` so we would need to accept `&[Hash]` as an input on the other function

Reply:
I've changed the logic though to just accept `&[Hash]` I think with current approach hashes were actually computed twice (if you look at `Clone` implementations of `Map` for instance it basically clones the input collection and applies the same Fn).